src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey instanceof Herbivore && !prey.hidden) {
+      prey.health -= 50;
+
+      prey.checkHealth();
```

Comment:
This is not a good practice to use the `checkHealth` method on the `prey` object. The `checkHealth` method is a part of the `Animal` class and it should be used on the `Animal` class itself. Instead, you can filter the `Animal.alive` array directly in the `bite` method after the `prey` health is decreased. This way you can ensure that only animals with health greater than 0 are left in the `Animal.alive` array.